### PR TITLE
docs: explain how to make LLVM builds that match emsdk's.

### PR DESCRIPTION
**Pull request by mr-c**
Inspired by my experience git-bisecting LLVM and it not working due to lack of `-DLLVM_ENABLE_ASSERTIONS=ON `.

### CODE REVIEW

File: site/source/docs/building_from_source/index.rst [hunk 1]
```diff
@@ -34,10 +34,14 @@ For example, something like this can work:
 
       mkdir build
       cd build/
-      cmake ../llvm -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='lld;clang' -DLLVM_TARGETS_TO_BUILD="host;WebAssembly" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF
+      cmake ../llvm -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='lld;clang' -DLLVM_TARGETS_TO_BUILD="host;WebAssembly" -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DLLVM_ENABLE_ASSERTIONS=ON
```
Comment (kripken) [inline site/source/docs/building_from_source/index.rst:37]: How about making assertions a comment afterwards? I think in general most users wouldn't want that overhead.

(In the emsdk itself we do it on normal builds but not on LTO builds, which are what we ship in most releases, iirc - @dschuff is that right?)

File: site/source/docs/building_from_source/index.rst [hunk 2]
```diff
@@ -34,10 +34,14 @@ For example, something like this can work:
...
       cmake --build .
 
-Then point LLVM_ROOT in ``.emscripten`` to ``<llvm_src>/build/bin`` (no need to install).
+Then set the environment variable ``EM_LLVM_ROOT`` to ``<llvm_src>/build/bin`` (no need to install).
+
+If you need to match the emsdk releases of LLVM, `review the emscripten-release
+build and test scripts <https://chromium.googlesource.com/emscripten-releases/+/refs/heads/main#build-and-test-scripts-in>`.
```
Comment (kripken) [inline site/source/docs/building_from_source/index.rst:43]:
```suggestion
build and test scripts <https://chromium.googlesource.com/emscripten-releases/+/refs/heads/main#build-and-test-scripts-in>`_.
```

File: site/source/docs/building_from_source/index.rst [hunk 3]
```diff
@@ -34,10 +34,14 @@ For example, something like this can work:
...
+Specifically `src/build.py <https://chromium.googlesource.com/emscripten-releases/+/refs/heads/main/src/build.py>`.
```
Comment (kripken) [inline site/source/docs/building_from_source/index.rst:44]:
```suggestion
Specifically `src/build.py <https://chromium.googlesource.com/emscripten-releases/+/refs/heads/main/src/build.py>`_.
```

File: site/source/docs/building_from_source/index.rst (hunk 2 above)
Comment (kripken) [reply site/source/docs/building_from_source/index.rst:43]: (sphinx notation is a little surprising...)

**Review by kripken — COMMENTED** [review]
Thanks!

File: site/source/docs/building_from_source/index.rst (hunk 1 above)
Comment (mr-c) [reply site/source/docs/building_from_source/index.rst]: Done!

**Review by kripken — APPROVED** [review]

**Comment by dschuff** [comment]
On a related note, I noticed that there's no mention on https://emscripten.org/docs/tools_reference/emsdk.html of using the emsdk tool to get assertions-enabled builds. Maybe we should fix that too (@mr-c I know that's not the use case you had... but were you aware of that feature?)

**Comment by mr-c** [comment]
> On a related note, I noticed that there's no mention on https://emscripten.org/docs/tools_reference/emsdk.html of using the emsdk tool to get assertions-enabled builds. Maybe we should fix that too (@mr-c I know that's not the use case you had... but were you aware of that feature?)

I eventually saw that, but because I was doing a git bisect, I wanted to build from a local checkout.

This whole issue came up because we (SIMDe) test using `tot` which I guess is not an LTO release.

**Comment by dschuff** [comment]
Ah yes, that's correct.

For future reference, I also just remembered that emsdk can help with bisecting too: https://emscripten.org/docs/contributing/developers_guide.html#bisecting

**Comment by mr-c** [comment]
> For future reference, I also just remembered that emsdk can help with bisecting too: https://emscripten.org/docs/contributing/developers_guide.html#bisecting

Yep, that's where [my journey started](https://github.com/emscripten-core/emscripten/issues/20505#issuecomment-1773713684) but then I needed to know which exact llvm commit was the cause.

**Pull request merged**